.github/workflows/poetry-tests.yaml

```diff
@@ -0,0 +1,56 @@
+name: Tests
+
+on:
+  workflow_call:
+    inputs:
+      python-version:
+        default: "3.11"
+        type: string
+      poetry-version:
+        default: "1.8.3"
+        type: string
+      os-versions:
+        default: "ubuntu-latest,macos-13,macos-14,windows-latest"
+        type: string
+      backends:
+        default: "Stubbed,QX emulator"
+        type: string
+    secrets:
+      e2e_username:
+        description: "E2E username"
+        required: true
+      e2e_password:
+        description: "E2E password"
+        required: true
+
+jobs:
+  lint:
+    uses: qutech-delft/github-actions/.github/workflows/_tox-lint.yaml@master
+    with:
+      python-version: ${{ inputs.python-version }}
+      poetry-version: ${{ inputs.poetry-version }}
+  unit-test:
+    uses: qutech-delft/github-actions/.github/workflows/_tox-unit-test.yaml@master
+    needs: [lint]
+    with:
+      python-version: ${{ inputs.python-version }}
+      poetry-version: ${{ inputs.poetry-version }}
+      os-versions: ${{ inputs.os-versions }}
+  e2e-test:
+    uses: qutech-delft/github-actions/.github/workflows/_custom-e2e-test.yaml@master
+    needs: [lint]
```

Comment:
IMO we should be running E2E test and unit tests parallel with lint. I presume you made them sequential to allow lint to find issues through static analysis that could also break the unit tests, but in practice 95% of the issues the linting tools find are formatting/typing stuff that don't impact unit tests. So I would say the speed up of running them all in parallel is worth it.

Reply:
The reason that we have lint as a first stage for all of our repositories, has to do with the duration of the unit tests. The linting is usually quite quick, and catches some of the obvious mistakes. The unit tests run on a number of environments, so take a long time (or cost more resources). Having the linting first is from the perspective of "fail fast"

Reply:
I'm aware of the trade off with resources, but I wouldn't consider (what we call) the linting step to be quick in general. In platform it takes just a bit more time than pleasant. As for "fail fast", if you execute the linting step in parallel with the tests you can detect a failure just as quickly. And then you get even more information as you can also find faults in the unit tests and E2E tests. Either way, this not a huge deal, so I am willing to approve regardless, just consider it. Feel free to resolve this conversation by yourself if you don't want to run them in parallel.

Reply:
Updated to run in parallel. Can you approve again?